src/internal/ui/sidebar/pinned_test.go

```diff
@@ -0,0 +1,309 @@
+package sidebar
+
+import (
+	"encoding/json"
+	"errors"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Load(t *testing.T) {
+	tempDir := t.TempDir()
+	pDirName := "pinnedDir"
+	pinnedDir := filepath.Join(tempDir, pDirName)
+	err := os.Mkdir(pinnedDir, 0755)
+	require.NoError(t, err)
+
+	emptyData := []directory{}
+	emptyBytes, err := json.Marshal(emptyData)
+	require.NoError(t, err)
+	emptyPath := filepath.Join(pinnedDir, "empty.json")
+	err = os.WriteFile(emptyPath, emptyBytes, 0644)
+	require.NoError(t, err)
+
+	invalidPath := filepath.Join(pinnedDir, "invalid.json")
+	err = os.WriteFile(invalidPath, []byte("{ invalid json }"), 0644)
+	require.NoError(t, err)
+
+	validData := []directory{
+		{
+			Location: pinnedDir,
+			Name:     pDirName,
+		},
+	}
+	validBytes, err := json.Marshal(validData)
+	require.NoError(t, err)
+	newValidPath := filepath.Join(pinnedDir, "valid.json")
+	err = os.WriteFile(newValidPath, validBytes, 0644)
+	require.NoError(t, err)
+
+	nonexistData := []directory{
+		{
+			Location: pinnedDir,
+			Name:     pDirName,
+		},
+		{
+			Location: filepath.Join(pinnedDir, "nonexistent9"),
+			Name:     "nonexistent9",
+		},
+	}
+	nonexistBytes, err := json.Marshal(nonexistData)
+	require.NoError(t, err)
+	newNonexistentPath := filepath.Join(pinnedDir, "nonexistent.json")
+	err = os.WriteFile(newNonexistentPath, nonexistBytes, 0644)
+	require.NoError(t, err)
+
+	cleanDirs := []directory{
+		{
+			Location: pinnedDir,
+			Name:     pDirName,
+		},
+	}
+
+	testCases := []struct {
+		name      string
+		pinnedMgr PinnedManager
+		expected  []directory
+	}{
+		{
+			name:      "Empty No Pinned Directories",
+			pinnedMgr: PinnedManager{filePath: emptyPath},
+			expected:  []directory{},
+		},
+		{
+			name:      "Invalid Format File",
+			pinnedMgr: PinnedManager{filePath: emptyPath},
+			expected:  []directory{},
+		},
+		{
+			name:      "Valid With No Non-Existent Directories",
+			pinnedMgr: PinnedManager{filePath: newValidPath},
+			expected:  cleanDirs,
+		},
+		{
+			name:      "Valid With One Non-Existent Directory",
+			pinnedMgr: PinnedManager{filePath: newNonexistentPath},
+			expected:  cleanDirs,
+		},
+	}
```

Comment:
Also add a test with a path of file that is invalid

something like

```suggestion
		{
			name:      "Valid With One Non-Existent Directory",
			pinnedMgr: PinnedManager{filePath: newNonexistentPath},
			expected:  cleanDirs,
		},
	},
	{
			name:      "Invalid filePath",
			pinnedMgr: PinnedManager{filePath: filepath.Join(pinnedDir, "non_existing_file_path.json")},
			expected: []directory{},
		},
	}
```